isobus/include/isobus/isobus/isobus_virtual_terminal_client.hpp

```diff
@@ -290,146 +292,170 @@ namespace isobus
-		void register_vt_soft_key_event_callback(VTKeyEventCallback value);
-
-		/// @brief Allows you to remove a callback for when a softkey is pressed or released
-		/// @param[in] value The callback to remove
-		void remove_vt_soft_key_event_callback(VTKeyEventCallback value);
-
-		/// @brief Allows you to register for a callback when a button is pressed or released
-		/// @param[in] value The callback to register
-		void register_vt_button_event_callback(VTKeyEventCallback value);
-
-		/// @brief Allows you to remove a callback when a button is pressed or released
-		/// @param[in] value The callback to remove
-		void remove_vt_button_event_callback(VTKeyEventCallback value);
-
-		/// @brief Allows you to register for a callback when a pointing event is "pressed or released"
-		/// @param[in] value The callback to register
-		void register_vt_pointing_event_callback(VTPointingEventCallback value);
-
-		/// @brief Allows you to remove a callback when a pointing event is "pressed or released"
-		/// @param[in] value The callback to remove
-		void remove_vt_pointing_event_callback(VTPointingEventCallback value);
-
-		/// @brief Allows you to register for a callback when an input object event is triggered
-		/// @param[in] value The callback to register
-		void register_vt_select_input_object_event_callback(VTSelectInputObjectCallback value);
-
-		/// @brief Allows you to remove an input object event callback
-		/// @param[in] value The callback to remove
-		void remove_vt_selection_input_object_event_callback(VTSelectInputObjectCallback value);
-
-		/// @brief Allows you to register for a callback when an ESC message is received, e.g. an open object input is closed
-		/// @param[in] value The callback to register
-		void register_vt_esc_message_event_callback(VTESCMessageCallback value);
-
-		/// @brief Allows you to remove an ESC message callback
-		/// @param[in] value The callback to remove
-		void remove_vt_esc_message_event_callback(VTESCMessageCallback value);
-
-		/// @brief Allows you to register for a callback when a numeric value is changed in an input object
-		/// @param[in] value The callback to register
-		void register_vt_change_numeric_value_event_callback(VTChangeNumericValueCallback value);
-
-		/// @brief Allows you to remove a numeric value change callback
-		/// @param[in] value The callback to remove
-		void remove_vt_change_numeric_value_event_callback(VTChangeNumericValueCallback value);
-
-		/// @brief Allows you to register for a callback when the active mask is changed
-		/// @details The VT sends this whenever there are missing object references or errors in the mask.
-		/// @param[in] value The callback to register
-		void register_vt_change_active_mask_event_callback(VTChangeActiveMaskCallback value);
+		/// @brief A struct for storing information of a VT key input event
+		struct VTKeyEvent
+		{
+			KeyActivationCode keyEvent; ///< The key event
+			std::uint8_t keyNumber; ///< The key number
```

Comment:
This is like, the most tiny, minor nit, but for all these structs, putting the uint8_t last may reduce the amount of memory padding that is needed for this struct. Likewise, putting the pointer first will also optimize this a little.

Reply:
Hmm, I think https://github.com/ad3154/Isobus-plus-plus/pull/211/commits/b2040cdeb98f58b46b8c1fd09113e85b0562397c resolves this? I am new to this memory padding stuff, and online they are rather vage about it. It seems like it's not really a standardized thing and the implementation differs across compilers and platforms? 

Though changing the order does make it a little harder to read I feel like, but since the structs are small and thus fast to read it's fine I think

Reply:
Yeah, good enough I think. The padding bytes depend a fair bit on architecture, mattering most to align memory addressing. So like, if your memory is byte addressable it doesn't matter much, but if it's word addressed and you have 32 bit words, then putting a u8 before a u32 may create 3 bytes of padding, for example, or the compiler (or even the microcode) could read both words and stitch them together, but that is not very efficient. I would hope that a smart compiler would re-order the struct in memory if optimizations are enabled and it thinks it can do so safely, but you never know.

Reply:
Looks like maybe the build is mad now due to the initializer list construction order changing

Reply:
Ohh yea, I forgot that. The build error is fixed in the latest commit🙉